tests/test_exchange_calendar.py

```diff
@@ -947,15 +950,19 @@ def _get_normal_session_block(self) -> pd.DatetimeIndex:
         """Block of 3 sessions with unchanged timings."""
         start_idx = len(self.sessions) // 3
         end_idx = start_idx + 21
+
+        def times_equal(*s):
+            for elems in zip(*s):
+                if not all(map(pd.isna, elems)):
```

Comment:
👍 Definitely cleaner.

Set comprehension might make it cleaner still?...
```python
if not set(elems) == {pd.NaT}:
    if not len({elem.time() for elem in elems}) == 1
```

Reply:
Yep definitely